nilearn/plotting/tests/test_displays.py

```diff
@@ -2,142 +2,189 @@
+)
 from nilearn.datasets import load_mni152_template
 
-##############################################################################
-# Some smoke testing for graphics-related code
 
-
-def test_demo_ortho_slicer():
-    # This is only a smoke test
-    oslicer = OrthoSlicer(cut_coords=(0, 0, 0))
-    img = load_mni152_template()
-    oslicer.add_overlay(img, cmap=plt.cm.gray)
-    oslicer.title("display mode is ortho")
-    oslicer.close()
-
-
-def test_demo_tiled_slicer():
-    tslicer = TiledSlicer(cut_coords=(0, 0, 0))
-    img = load_mni152_template()
-    tslicer.add_overlay(img, cmap=plt.cm.gray)
-    tslicer.close()
-
-
-def test_stacked_slicer():
-    # Test stacked slicers, like the XSlicer
-    img = load_mni152_template()
-    slicer = XSlicer.init_with_figure(img=img, cut_coords=3)
+SLICER_KEYS = [
+    'ortho', 'tiled', 'x', 'y', 'z', 'yx', 'yz', 'mosaic'
+]
+SLICERS = [
+    OrthoSlicer, TiledSlicer, XSlicer, YSlicer, ZSlicer,
+    YXSlicer, YZSlicer, MosaicSlicer
+]
+PROJECTOR_KEYS = [
+    'ortho', 'xz', 'yz', 'yx', 'lyrz', 'lyr', 'lzr', 'lr', 'l', 'r'
+]
+PROJECTORS = [
+    OrthoProjector, XZProjector, YZProjector, YXProjector, XProjector,
+    YProjector, ZProjector, LZRYProjector, LYRZProjector, LYRProjector,
+    LZRProjector, LRProjector, LProjector, RProjector
+]
+
+
+@pytest.fixture
+def img():
+    """Image used for testing."""
+    return load_mni152_template()
+
+
+@pytest.fixture
+def cut_coords(name):
+    """Selects appropriate cut coords."""
+    if name == 'mosaic': return 3
+    if name in ['yx', 'yz', 'xz']: return (0,) * 2
+    if name in ['lyrz', 'lyr', 'lzr']: return (0,)
+    if name in ['lr', 'l']: return (0,) * 4
+    return (0,) * 3
+
+
+@pytest.mark.parametrize("display,name",
+                         zip(SLICERS + PROJECTORS,
+                             SLICER_KEYS + PROJECTOR_KEYS))
+def test_display_basics(display, name, img, cut_coords):
+    """Basic smoke tests for all displays (slicers + projectors).
+    Each object is instanciated, ``add_overlay``, ``title``,
+    and ``close`` are then called.
+    """
+    display = display(cut_coords=cut_coords)
+    display.add_overlay(img, cmap=plt.cm.gray)
+    display.title(f"display mode is {name}")
+    if name != "mosaic":
+        assert display.cut_coords == cut_coords
+    assert isinstance(display.frame_axes, matplotlib.axes.Axes)
+    display.close()
+
+
+@pytest.mark.parametrize("slicer",
+                         [XSlicer, YSlicer, ZSlicer, YXSlicer, YZSlicer])
+def test_stacked_slicer(slicer, img, tmp_path):
+    """Tests for saving to file with stacked slicers."""
+    cut_coords = (3, 3) if slicer in [YXSlicer, YZSlicer] else 3
+    slicer = slicer.init_with_figure(img=img, cut_coords=cut_coords)
     slicer.add_overlay(img, cmap=plt.cm.gray)
     # Forcing a layout here, to test the locator code
-    with tempfile.TemporaryFile() as fp:
-        slicer.savefig(fp)
+    slicer.savefig(tmp_path / "out.png")
```

Comment:
just curious: why do you replace this ?

Reply:
that is old code, from before we were using pytest and the `tmp_path` fixture was available -- in new tests, we would use `tmp_path` as it is a bit simpler. the older way, using `tempfile`, still works fine though